### ISSUE TITLE
Qt Compatibility Error (Maya 2018)

### ISSUE THREAD

**Issue by valmikchavan** [david-cattermole/mayaMatchMoveSolver#88]
HI David,

i have copy the files into maya 2018 modules folder, but the solve UI is not getting open. getting below error,

# Error: TypeError: file /home/valmik.chavan/maya/2018/modules/mayaMatchMoveSolver-0.2.0-maya2018-linux/python/mmSolver/ui/uiutils.py line 64: You need a shiboken-based type.

Thanks, 
valmik

**Comment by david-cattermole** [comment]
Hello Valmik,

After doing a quick search I believe you have PyQt installed in Maya, which I am not expecting. mmSolver is tested using the standard Maya installation.

For example, here is the same problem:
https://stackoverflow.com/questions/31117393/you-need-a-shiboken-based-type

I will attempt to find a way to fix this, if so, it will be included in the v0.2.1 release.

David

**Comment by david-cattermole** [comment]
Hello again,

After searching more I believe I may have found the issue, but I cannot test it because I do not have PyQt installed (which I am guessing is causing the problem).

Following the advice from this page:
https://forums.autodesk.com/t5/maya-programming/pyside-maya-openmayaui-mqtutil-mainwindow-returning-swigpyobject/td-p/4574059

I believe we might fix the problem by changing line 64 in "/home/valmik.chavan/maya/2018/modules/mayaMatchMoveSolver-0.2.0-maya2018-linux/python/mmSolver/ui/uiutils.py":
From this:
```
    window = wrapInstance(long(window_ptr), QtWidgets.QWidget)
```

To this:
```
    window = wrapInstance(long(window_ptr), QtWidgets.QMainWindow)
```
Notice the "QWidget" has changed to "QMainWindow".

Valmik, can you please change this line number 64 in the file `uiutils.py`, then reply with the results if it fixed the problem or not?

If you are working in a professional environment (studio), I'd recommend asking your pipeline, systems or technology department to install mmSolver, rather than in your home directory.

Thanks for reporting this!

David

**Comment by valmikchavan** [comment]
Hi David,

Thanks for the update, we'll try out the steps you mention, also we'll get
back to you if we are able to get results.

Thanks again.

**Comment by david-cattermole** [comment]
Thanks Valmik,

For the v0.2.1 release, I'll include whatever fixes I think will fix the problem for you.
If I get some time, I will install PyQt into Maya as well and try to re-create your environment to reproduce the problem.

David

**Comment by david-cattermole** [comment]
Hello Valmik,

The new v0.2.1 should hopefully address the bug you've found.
I have added a few attempts to fix the bug, not just editing line 64 in uiutils.py.

Can you try installing v0.2.1 and let me know how it goes?

https://github.com/david-cattermole/mayaMatchMoveSolver/releases/tag/v0.2.1

David

**Comment by david-cattermole** [comment]
The fix in v0.2.1 has used the module ``Qt.QtCompat``.

A user is using Qt.py/1.1.0-b7.

The ``Qt.QtCompat`` module will fail to import using Qt.py/1.1.0-b7, when using:
```
import Qt.QtCompat
````

However ``Qt.QtCompat`` can be used in Qt.py/1.1.0-b7, like this: 
```
import Qt
print Qt.QtCompat
```

We should make v0.2.2 compatible with Qt.py/1.1.0-b7.

**Comment by valmikchavan** [comment]
Hello David,

Yeah, it's working as expected now, thank you very much!

Have a great weekend :)

**Comment by david-cattermole** [comment]
Thanks for letting me know Valmik!

There will be a minor change to be released in v0.2.2. This change should not affect your problem at all, it's just for backwards compatibility. Let me know if it stops working for you.

I'm going to close this issue now.

David
